Managers/MapManager.gd

```diff
@@ -292,3 +292,11 @@ func debug_print_event_count() -> void:
 	for k: String in events:
 		print("Event " + k + " has " + str(events[k]) + " rooms (expected: "+ str(float(expected_probabilities[k] * total_nb_rooms/100.0)).pad_decimals(2) +"). (The percentage is " + str(float(events[k]) * 100 / total_nb_rooms).pad_decimals(2) + "%, expected: " + str(expected_probabilities[k]) + "%)")
 	print("Total number of rooms generated: " + str(total_nb_rooms))
+	
+
+## Returns the percent height of the player in the current map
+func get_map_percent_with_player_position() -> float:
+	var player_position: Vector2i = PlayerManager.player_position
+	var number_of_floors: int = map_width_array.size()
+	var player_y_floor: int = player_position.y
+	return (player_y_floor/number_of_floors) * 100
```

Comment:
(player_y_floor/number_of_floors) is always returning 0
Both variables needs to be floats insetad of int (or cast to floats at division)

Reply:
mmh that's strange, i'll look into it

Reply:
Fixed in 62734436aac9

Reply:
I was also thinking we could do `(player_y_floor *100 / number_of_floors as float)` instead of casting both to float

Reply:
That could also work - there would however be a difference in cases where we hit decimals like 19.853% would get rounded to 20 and fall into 20-40 distribution, whereas with current Float casting it would fall under 0-20.